### ISSUE TITLE
Is the concurrency under this architecture subject to molecular rather than meteor?

### ISSUE THREAD

**Issue by dzmazhuo** [RocketChat/Rocket.Chat#28269]
![microservice](https://1635978725-files.gitbook.io/~/files/v0/b/gitbook-x-prod.appspot.com/o/spaces%2F-M418Ul0aSTwf2PYsyPW%2Fuploads%2Fgit-blob-b43b8ed3175cfa15cc8a6a4affcfc45be74342bd%2FMicro%20services%20deployment%20-%20v0.1%402x%20(1).png?alt=media)

The picture above shows the architecture of Microservices in RocketChat. I know what the services(DDP streamer, account, Authorization, Presence, Stream hub) means that they the microservices of molecular. 
My question is :
- what is Rocket.Chat(monolith)?
- Is the concurrency under this architecture subject to molecular rather than meteor?

**Comment by seconp** [comment]
No, I assume that ws connection is under molecularjs while http connection is under meteor. Hign concurrency of http is easy to resolve, but high concurrency of ws is difficult. I suggest u build your own middleware with C languge. Then with load balancing, you can achieve hign concurrency.
